### PR TITLE
CASSANDRA-17009 Adds unit tests to sstableverify operate on SSTables

### PR DESCRIPTION
**Pull request by bhouse99**
sstableverify tests didn't cover the case of exercising the verifyier for sstables.
Added unit tests to verify this and several corrupt scenarios.

test/data was marked as git ignore, despite it containing files for use.
Added an exception to the gitignore so forcing test data files is no longer necessary.

There are some small bugs found which will be in subsequent jiras.
(Return codes being wrong, etc).

    - Jira : https://issues.apache.org/jira/browse/CASSANDRA-17009

### CODE REVIEW

File: test/unit/org/apache/cassandra/tools/StandaloneVerifierOnSSTablesTest.java [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+            {
+                file.seek(0);
+                file.writeBytes(StringUtils.repeat('z', 2));
+            }
+
+        });
+
+        ToolResult tool = ToolRunner.invokeClass(StandaloneVerifier.class, keyspaceName, corruptStatsTable);
+
+        assertEquals(0, tool.getExitCode());
```
Comment (bereng) [inline test/unit/org/apache/cassandra/tools/StandaloneVerifierOnSSTablesTest.java]: Note to self: Should we return 1 here?

File: test/unit/org/apache/cassandra/tools/StandaloneVerifierOnSSTablesTest.java [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+    @Test
+    public void testWorkingDataFile() throws Exception
+    {
+        String keyspaceName = "StandaloneVerifierTestWorkingDataKs";
+        String workingTable = "workingTable";
+
+        createAndPopulateTable(keyspaceName, workingTable, x -> {});
+
+        ToolResult tool = ToolRunner.invokeClass(StandaloneVerifier.class, keyspaceName, workingTable);
+        assertEquals(0, tool.getExitCode());
```
Comment (bereng) [inline test/unit/org/apache/cassandra/tools/StandaloneVerifierOnSSTablesTest.java]: `assertOnCleanExit()`?

File: test/unit/org/apache/cassandra/tools/StandaloneVerifierOnSSTablesTest.java [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+        assertCorrectEnvPostTest();
+    }
+
+    /**
+     * Since we are testing a verifier, we'd like to corrupt files to verify code paths
+     * This function defiintion is used by {@link this#createAndPopulateTable}.
```
Comment (bereng) [inline test/unit/org/apache/cassandra/tools/StandaloneVerifierOnSSTablesTest.java]: defiinition -> definition

File: test/unit/org/apache/cassandra/tools/StandaloneVerifierOnSSTablesTest.java [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+     *
+     * CFS is the open ColumnFamilyStore for a given keyspace, table
+     */
+    private interface CorruptFunction {
+        public void apply(ColumnFamilyStore cfs) throws Exception;
+    }
+
+    /**
+     * This function sets up the keyspace, and table schema for a standardCFMD table.
+     * This will also populate the tableiName with a few rows.  After completion the
```
Comment (bereng) [inline test/unit/org/apache/cassandra/tools/StandaloneVerifierOnSSTablesTest.java]: double space before `After`

File: test/unit/org/apache/cassandra/tools/StandaloneVerifierOnSSTablesTest.java [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+     * server will be shutdown.
+     *
+     * @param keyspace name the table should be created in
+     * @param tableName new table name of the standard cfmd table
+     *
+     * @param corruptionFn function called to corrupt or change the contents on disk.  is pased the Cfs of the table name
```
Comment (bereng) [inline test/unit/org/apache/cassandra/tools/StandaloneVerifierOnSSTablesTest.java]: Typos

File: test/unit/org/apache/cassandra/tools/StandaloneVerifierTest.java [hunk 6]
```diff
@@ -153,4 +153,13 @@ public void testVerboseArg()
...
     }
+
+    @Test
+    public void testTooManyArgs()
+    {
+        ToolResult tool = ToolRunner.invokeClass(StandaloneVerifier.class, "another arg", "system_schema", "tables");
+        assertThat(tool.getStdout(), CoreMatchers.containsStringIgnoringCase("usage:"));
+        assertThat(tool.getCleanedStderr(), CoreMatchers.containsStringIgnoringCase("Too many arguments"));
+        assertEquals(1, tool.getExitCode());
+    }
```
Comment (bereng) [inline test/unit/org/apache/cassandra/tools/StandaloneVerifierTest.java:167]: While we are here I'd look for opportunities to use `assertOnCleanExit()` in this file

File: test/unit/org/apache/cassandra/tools/StandaloneVerifierOnSSTablesTest.java (hunk 1 above)
Comment (bereng) [reply test/unit/org/apache/cassandra/tools/StandaloneVerifierOnSSTablesTest.java]: No it's a bug

Comment (bhouse99) [reply test/unit/org/apache/cassandra/tools/StandaloneVerifierOnSSTablesTest.java]: fixed

File: test/unit/org/apache/cassandra/tools/StandaloneVerifierOnSSTablesTest.java (hunk 2 above)
Comment (bhouse99) [reply test/unit/org/apache/cassandra/tools/StandaloneVerifierOnSSTablesTest.java]: fixed

File: test/unit/org/apache/cassandra/tools/StandaloneVerifierOnSSTablesTest.java (hunk 3 above)
Comment (bhouse99) [reply test/unit/org/apache/cassandra/tools/StandaloneVerifierOnSSTablesTest.java]: fixed

File: test/unit/org/apache/cassandra/tools/StandaloneVerifierOnSSTablesTest.java (hunk 4 above)
Comment (bhouse99) [reply test/unit/org/apache/cassandra/tools/StandaloneVerifierOnSSTablesTest.java]: fixed

File: test/unit/org/apache/cassandra/tools/StandaloneVerifierOnSSTablesTest.java (hunk 5 above)
Comment (bhouse99) [reply test/unit/org/apache/cassandra/tools/StandaloneVerifierOnSSTablesTest.java]: fixed

File: test/unit/org/apache/cassandra/tools/StandaloneVerifierTest.java (hunk 6 above)
Comment (bhouse99) [reply test/unit/org/apache/cassandra/tools/StandaloneVerifierTest.java:167]: Replaced on all that could have it run

File: test/unit/org/apache/cassandra/tools/StandaloneVerifierOnSSTablesTest.java [hunk 7]
```diff
@@ -0,0 +1,220 @@
...
+                file.writeBytes(StringUtils.repeat('z', 2));
+            }
+        });
+
+        ToolResult tool = ToolRunner.invokeClass(StandaloneVerifier.class, keyspaceName, corruptDataTable);
+        assertEquals(1, tool.getExitCode());
+        Assertions.assertThat(tool.getStdout()).contains("Invalid SSTable", corruptDataTable);
+    }
+
+
```
Comment (bereng) [inline test/unit/org/apache/cassandra/tools/StandaloneVerifierOnSSTablesTest.java:165]: Extra blanks

File: test/unit/org/apache/cassandra/tools/StandaloneVerifierTest.java [hunk 8]
```diff
@@ -70,6 +70,8 @@ public void testDefaultCall()
         Assertions.assertThat(tool.getCleanedStderr()).isEmpty();
         assertEquals(0, tool.getExitCode());
         assertCorrectEnvPostTest();
+        tool.assertOnCleanExit();
```
Comment (bereng) [inline test/unit/org/apache/cassandra/tools/StandaloneVerifierTest.java:73]: This test file doesn't mention the other two and vice-versa

**Pull request closed without merging**